### ISSUE TITLE
Recursive copy from GCS bucket to Azure container fails when using `exclude-path` with `cannot start job due to error: FileIgnored` error

### ISSUE THREAD

**Issue by pstibrany** [Azure/azure-storage-azcopy#1640]
### Which version of the AzCopy was used? 

`AzCopy 10.13.0`

### Which platform are you using? (ex: Windows, Mac, Linux)

Mac

### What command did you run?

```
azcopy copy 'https://storage.cloud.google.com/bucket/test' "https://account.blob.core.windows.net/container/test" --recursive --exclude-pattern bucket-index.json.gz
```

### What problem was encountered?

```
failed to perform copy command due to error: cannot start job due to error: FileIgnored.
```

### How can we reproduce the problem in the simplest way?

Whenever I try to do recursive copy from Google Storage bucket and use `--exclude-*` flags (tried `--exclude-path`, `--exclude-regex` and `--exclude-pattern`) to skip some files, I get the error `failed to perform copy command due to error: cannot start job due to error: FileIgnored.`.

### Have you found a mitigation/solution?

Removing exclusion works fine, but of course also copies objects that I would prefer not to copy.

**Comment by adreed-msft** [comment]
This is an error that was supposed to be caught and tracked elsewhere; it should not have made it here. I've fixed this in #1654 and it should hopefully make it through next release.

**Comment by pstibrany** [comment]
Thank you! I will test it and report back.

**Comment by pstibrany** [comment]
I confirm that with #1654 the exclude options work fine. Thank you very much!

**Comment by mohsha-msft** [comment]
Fix released in AzCopy `10.14.0`
